docs/proposals/nginx-extensions.md

```diff
@@ -0,0 +1,964 @@
+| Client keepalives                                                                                |                     |
+| TLS settings. For example, TLS protocols and server ciphers                                      |                     |
+| OpenTelemetry tracing                                                                            |                     |
+
+### Medium-Priority Features
+
+| Features                                                                               | Requires NGINX Plus |
+|----------------------------------------------------------------------------------------|---------------------|
+| Circuit breaker                                                                        |                     |
+| Load-balancing method                                                                  |                     |
+| Load-balancing method (least time)                                                     | X                   |
+| Limit connections to a server                                                          |                     |
+| Request queuing. Queue requests if servers are unavailable                             | X                   |
+| Basic authentication. User/Password                                                    |                     |
+| PROXY protocol                                                                         |                     |
+| Upstream zone size                                                                     |                     |
+| Custom response. Return a custom response for a given path                             |                     |
+| External routing. Route to services outside the cluster                                |                     |
+| HTTP Strict Transport Security (HSTS)                                                  |                     |
+| API Key authentication                                                                 |                     |
+| Next upstream retries. NGINX retries by trying the _next_ server in the upstream group |                     |
+| Proxy Buffering                                                                        |                     |
+| Connection timeouts                                                                    |                     |
+
+### Low-Priority Features
+
+| Features                                   | Requires NGINX Plus |
+|--------------------------------------------|---------------------|
+| Pass/hide headers to/from the client       |                     |
+| Custom error pages                         |                     |
+| Access control                             |                     |
+| Windows Technology LAN Manager (NTLM)      | X                   |
+| Backend client certificate verification    |                     |
+| NGINX worker settings                      |                     |
```

Comment:
by default in our config https://github.com/nginxinc/nginx-gateway-fabric/blob/de23bce0409b91cefa4e01bc5785f07d55eac4ee/internal/mode/static/nginx/conf/nginx.conf#L3 NGINX will span as many workers as cores. This means, if user has 16 or 32 core nodes, it will span 16 or 32 workers.
This might not be needed - recommendation is to use one worker per core but assuming the dedicated machine, not a shared node. At the same time, more workers, more overhead. 

I suggest we prioritize configuring this directive.

Reply:
@mpstefan ^

Reply:
Spoke offline, and for now, we will leave `worker_processes` in the low priority section. This is a performance optimization, and we are prioritizing features that enable use cases.